tests/functional/test_model_completeness.py

```diff
@@ -46,3 +48,48 @@ def test_paginators_and_waiters_are_not_lost_in_new_version(
             raise AssertionError(
                 f"{type_name} must exist for {service_name}: {e}"
             )
+
+
+def _endpoint_rule_set_cases():
+    for service_name in Session().get_available_services():
+        versions = Loader().list_api_versions(service_name, 'service-2')
+        for version in versions:
+            yield service_name, version
+
+
+@pytest.mark.parametrize(
+    "service_name, version",
+    _endpoint_rule_set_cases(),
+)
+def test_all_endpoint_rule_sets_exist(service_name, version):
+    """Tests the existence of endpoint-rule-set.json for each service
+    and verifies that content is present."""
+    loader = Loader()
+    type_name = 'endpoint-rule-set'
+    loader.load_service_model(service_name, type_name, version)
+    full_path = os.path.join(service_name, version, type_name)
+    data = loader.load_data(full_path)
+    assert len(data['rules']) >= 1
+
+
+test_data_dir = os.path.join(os.path.dirname(__file__), "endpoint-rules")
+
+
+def test_all_endpoint_tests_exist():
+    """Tests the existence of endpoint-tests.json for each service
+    and verifies that content is present."""
+    for service_name in Session().get_available_services():
+        file_name = 'endpoint-tests.json'
+        endpoint_tests_file = os.path.join(
+            test_data_dir, service_name, file_name
+        )
+        with open(endpoint_tests_file) as f:
+            data = json.load(f)
+            assert len(data['testCases']) >= 1
+
+
+def test_partitions_exists():
+    """Tests the existence of partitions.json and verifies that content is present."""
+    loader = Loader()
+    data = loader.load_data('partitions')
+    assert len(data['partitions']) >= 1
```

Comment:
I think we know there will be at least 4 partitions. Let's set our lower bound for now to avoid regressing. `[aws, aws-cn, aws-iso, aws-iso-b]`